website/content/en/docs/kubebuilder/references/logging.md

```diff
@@ -0,0 +1,262 @@
+
+* `--zap-devel`: Development Mode defaults(encoder=consoleEncoder,logLevel=Debug,stackTraceLevel=Warn)
+			  Production Mode defaults(encoder=jsonEncoder,logLevel=Info,stackTraceLevel=Error)
+* `--zap-encoder`: Zap log encoding ('json' or 'console')
+* `--zap-log-level`: Zap Level to configure the verbosity of logging. Can be one of 'debug', 'info', 'error',
+			       or any integer value > 0 which corresponds to custom debug levels of increasing verbosity")
+* `--zap-stacktrace-level`: Zap Level at and above which stacktraces are captured (one of 'warn' or 'error')
+
+Please refer [godocs][logging_godocs] for more detailed flag information.
+
+### A simple example
+
+Operators set the logger for all operator logging in [`cmd/manager/main.go`][code_set_logger]. To illustrate how this works, try out this simple example:
+
+```Go
+package main
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"  
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var globalLog = logf.Log.WithName("global")
+func main() {
+	// Add the zap logger flag set to the CLI. The flag set must
+	// be added before calling flag.Parse().
+	opts := zap.Options{}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+    
+	logger := zap.New(zap.UseFlagOptions(&opts))
+	logf.SetLogger(logger)
+
+	scopedLog := logf.Log.WithName("scoped")
+
+	globalLog.Info("Printing at INFO level")
+	globalLog.V(1).Info("Printing at DEBUG level")
+	scopedLog.Info("Printing at INFO level")
+	scopedLog.V(1).Info("Printing at DEBUG level")
+}
+```
+
+#### Output using the defaults
+```console
+$ go run main.go
+INFO[0000] Running the operator locally in namespace default. 
+{"level":"info","ts":1587741740.407766,"logger":"global","msg":"Printing at INFO level"}
+{"level":"info","ts":1587741740.407855,"logger":"scoped","msg":"Printing at INFO level"}
+```
+
+#### Output overriding the log level to 1 (debug)
+```console
+$ go run main.go --zap-log-level=debug
+INFO[0000] Running the operator locally in namespace default. 
+{"level":"info","ts":1587741837.602911,"logger":"global","msg":"Printing at INFO level"}
+{"level":"debug","ts":1587741837.602964,"logger":"global","msg":"Printing at DEBUG level"}
+{"level":"info","ts":1587741837.6029708,"logger":"scoped","msg":"Printing at INFO level"}
+{"level":"debug","ts":1587741837.602973,"logger":"scoped","msg":"Printing at DEBUG level"}
+```
+## Custom zap logger
+
+In order to use a custom zap logger, [`zap`][controller_runtime_zap] from controller-runtime can be utilized to wrap it in a `logr` implementation.
+
+Below is an example illustrating the use of [`zap-logfmt`][logfmt_repo] in logging.
+
+### Example
+
+In your `main.go` file, replace the current implementation for logs inside the `main` function:
+
+```Go
+import(
+	...
+	zaplogfmt "github.com/sykesm/zap-logfmt"
+	uzap "go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	...
+)
```

Comment:
`zaplogfmt `, `uzap`, and `zapcore` are the new imports used in the updated example. Not here in the current code.
Move this import block down into the the replacement code block. And remove it from here.
```suggestion
```